src/coreComponents/common/MpiWrapper.hpp

```diff
@@ -1115,6 +1125,30 @@ void MpiWrapper::reduce( Span< T const > const src, Span< T > const dst, Reducti
   reduce( src.data(), dst.data(), LvArray::integerConversion< int >( src.size() ), getMpiOp( op ), root, comm );
 }
 
+// Mpi helper function to return  tuple containing the max value of element 0 in the tuple if rank 0
+// For rank > 0 input tuple is returned
+template< typename tuple_type >
+tuple_type
+MpiWrapper::tupleMaxLoc( tuple_type tuple, MPI_Comm comm )
+{
+  // recieve buffer
+  std::vector< tuple_type > l_rcv_buffer;
+
+  int const myRank  = MpiWrapper::commRank( comm );
+  int const numProcs = MpiWrapper::commSize( comm );
+  if( myRank == 0 )
+  {
+    l_rcv_buffer.resize( numProcs );
+  }
+  MPI_Gather( &tuple, sizeof(tuple_type), MPI_BYTE, &l_rcv_buffer[0], sizeof(tuple_type), MPI_BYTE, 0, comm );
+
+
+  if( myRank == 0 )
+  {
+    tuple= *max_element( begin( l_rcv_buffer ), end( l_rcv_buffer ), []( auto & lhs, auto & rhs ) -> integer {return std::get< 0 >( lhs )  <  std::get< 0 >( rhs ); } );
+  }
+  return tuple;
```

Comment:
I think you should add `brodacast` step. I understand that you don't need it for this specific task but since you are creating a generic function I would expect it to also perform the broadcast.

Reply:
Either broadcast, or change the name of the function.